pkg/frontend/admin_vm_actions.go

```diff
@@ -0,0 +1,53 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/adminactions"
+)
+
+type azVmActionsWrapper struct {
+	vmName       string
+	adminActions adminactions.AzureActions
+	doc          *api.OpenShiftClusterDocument
+}
+
+func (f *frontend) newAzureActionsWrapper(ctx context.Context, vmName string, resourceID string, vars map[string]string, log *logrus.Entry) (azVmActionsWrapper, error) {
```

Comment:
I just realized that the logrus thing should probably be inside the frontend struct for this function and probably `f.azureActionsFactory` ( maybe others too), so we don't have to pass it around using parameters. Does it belong in this PR or another, I'm not sure, I'll let you be the judge of that.

A little nit while I'm at it, you can write `vmName , resourceID string` instead of writing string twice.

Reply:
+1 I think more can be done on the log side but it should be in a separate  PR